backend/src/libs/packages/mailer/mailer.package.ts

```diff
@@ -0,0 +1,50 @@
+import { default as aws, SESClient } from '@aws-sdk/client-ses';
+import nodemailer from 'nodemailer';
+import { type default as Mail } from 'nodemailer/lib/mailer/index.js';
+
+import { type IConfig } from '~/libs/packages/config/config.js';
+
+import { EmailFailedToSendError } from '../exceptions/exceptions.js';
+import { getResetPasswordEmailTemplate } from './libs/templates/templates.js';
+import { type SendEmailResponse } from './libs/types/types.js';
+
+class Mailer {
+  private sesClient: SESClient;
+  private config: IConfig;
+  private transporter: nodemailer.Transporter;
+
+  public constructor(config: IConfig) {
+    this.config = config;
+
+    const { AWS } = this.config.ENV;
+
+    this.sesClient = new SESClient({
+      region: AWS.AWS_REGION,
+      credentials: {
+        secretAccessKey: AWS.AWS_SECRET_ACCESS_KEY,
+        accessKeyId: AWS.AWS_ACCESS_KEY,
+      },
+    });
+    this.transporter = nodemailer.createTransport({
+      SES: { ses: this.sesClient, aws },
+    });
+  }
+
+  public async sendResetPasswordEmail(
+    recipient: string,
+    resetLink: string,
+  ): Promise<SendEmailResponse> {
+    const mail = getResetPasswordEmailTemplate(recipient, resetLink);
+    return await this.sendMail(mail);
+  }
+
+  public async sendMail(mail: Mail.Options): Promise<SendEmailResponse> {
+    try {
+      return (await this.transporter.sendMail(mail)) as SendEmailResponse;
+    } catch {
+      throw new EmailFailedToSendError();
```

Comment:
If transporter returns response from email API - then catch may not work (AWS likes to return errors in response), so let's add a notion that we may pay attention on it

Reply:
Can you clarify , please ? I didn't quite understand what I need to do.

Reply:
Just add a comment for this catch block